charts/nextcloud/1.6.31/templates/deployment.yaml

```diff
@@ -136,6 +136,21 @@ spec: {{ include "common.deployment.common_spec" . | nindent 2 }}
             httpHeaders:
             - name: Host
               value: localhost
+        {{- end }}
+        {{- if .Values.nextcloud.install_smbclient }}
+          exec:
+            command:
+              - /bin/sh
+              - -c
+              - |
+                command -v smbclient /dev/null 2>&1 || exit 1
+        {{- else }}
+          httpGet:
+            path: /status.php
+            port: http
+            httpHeaders:
+            - name: Host
+              value: localhost
         {{- end }}
```

Comment:
Probes can't have multiple types defined at the same time.
eg if none of those are enabled, you will end up with 2 httpGet in startupProbe

Logic for probe type selection and script under exec type have to be merged.